miso-web/src/main/java/uk/ac/bbsrc/tgac/miso/webapp/controller/EditFreezerController.java

```diff
@@ -0,0 +1,64 @@
+package uk.ac.bbsrc.tgac.miso.webapp.controller;
+
+import java.io.IOException;
+import java.util.List;
+import java.util.Map;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.security.acls.model.NotFoundException;
+import org.springframework.stereotype.Controller;
+import org.springframework.ui.ModelMap;
+import org.springframework.web.bind.annotation.ModelAttribute;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.servlet.ModelAndView;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+import uk.ac.bbsrc.tgac.miso.core.data.impl.StorageLocation;
+import uk.ac.bbsrc.tgac.miso.core.data.impl.StorageLocation.LocationUnit;
+import uk.ac.bbsrc.tgac.miso.dto.Dtos;
+import uk.ac.bbsrc.tgac.miso.service.StorageLocationService;
+
+@Controller
+@RequestMapping("/freezer")
+public class EditFreezerController {
+
+  private static final String JSP = "/pages/editFreezer.jsp";
+
+  private static final String MODEL_ATTR_PAGEMODE = "pageMode";
+  private static final String MODEL_ATTR_JSON = "freezerJson";
+
+  @Autowired
+  private StorageLocationService storageLocationService;
+
+  @ModelAttribute("maxLengths")
+  public Map<String, Integer> getColumnSizes() throws IOException {
+    return storageLocationService.getColumnSizes();
+  }
+
+  @ModelAttribute("rooms")
+  public List<StorageLocation> getRooms() {
+    return storageLocationService.listRooms();
+  }
+
+  @RequestMapping("/new")
+  public ModelAndView newArray(ModelMap model) throws IOException {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove the declaration of thrown exception 'java.io.IOException', as it cannot be thrown from method's body. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)